File: String/strStr.py
#
# Return index of first substring occurence
#


def strStr(haystack, needle):
    if len(needle) == 0:
        return 0

    for x in range(0, len(haystack)):
        if haystack[x] == needle[0]:
            if haystack[x : x + len(needle)] == needle:
                return x
    return -1


print(strStr("hello", "ll"))


# Time complexity: 0(N); space complexity: O(N);

File: String/strStr.js
/*
 * Return index of first substring occurence
 */

const strStr = (haystack, needle) => {
  if (needle.length === 0) return 0;

  for (i = 0; i < haystack.length; i++) {
    if (haystack[i] === needle[0]) {
      if (haystack.substr(i, needle.length) === needle) {
        return i;
      }
    }
  }

  return -1;
};

console.log(strStr("Hello", "ll"));

// * Time complexity: 0(N); space complexity: O(N);
